packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -230,6 +240,10 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     ? templateParams.updateExisting(finalParameterValues, cloudFormationStack.parameters)
     : templateParams.supplyAll(finalParameterValues);
 
+  if (options.hotswapLogMonitor) {
+    await registerCloudWatchLogGroups(options.sdkProvider, stackArtifact, assetParams, options.hotswapLogMonitor);
+  }
+
```

Comment:
A good question is whether we should register the log groups before the deployment, or after it.

If you do it before, like you do it here, I won't get the logs of a newly created Lambda function that I just added to my Stack, which is probably not a great experience.

Reply:
I agree. I moved this to run _after_ deploy.